src/Tools/RapidResponseDesk/sections/AreaTime/AreaAndTimeSection.jsx

```diff
@@ -1,18 +1,333 @@
+    tempTimespan,
+  ) => {
+    return (
+      tempTimespan.from.isBetween(selectedFromTimespan, selectedToTimespan, undefined, '[]') ||
+      tempTimespan.to.isBetween(selectedFromTimespan, selectedToTimespan, undefined, '[]')
+    );
+  };
+
+  const detectIfDateIsOverlapping = (id, selectedFromTimespan, selectedToTimespan) => {
+    const overlappedFromRange = timespanArray.filter((tempTimespan) => {
+      if (id !== tempTimespan.id) {
+        return (
+          checkIfSelectedTimespanIsBetweenTempTimespan(
+            selectedFromTimespan,
+            selectedToTimespan,
+            tempTimespan,
+          ) ||
+          checkIfTempTimespanIsBetweenSelectedTimespan(selectedFromTimespan, selectedToTimespan, tempTimespan)
+        );
+      } else {
+        return undefined;
+      }
+    });
+
+    const overlappedToRange = timespanArray.filter((currentTimespan) => {
+      if (id !== currentTimespan.id) {
+        return selectedToTimespan.isBetween(currentTimespan.from, currentTimespan.to, undefined, '[]');
+      } else {
+        return undefined;
+      }
+    });
+
+    let newOverlappedRanges = overlappedRanges ? [...overlappedRanges] : [];
+    storeRangesWhichAreOverlapping(overlappedFromRange, overlappedToRange, newOverlappedRanges, id);
+  };
+
+  const storeRangesWhichAreOverlapping = (
+    overlappedFromRange,
+    overlappedToRange,
+    newOverlappedRanges,
+    id,
+  ) => {
+    // clear all out for selected id and check for new overlaps
+    newOverlappedRanges = newOverlappedRanges.filter(
+      (range) => range.selectedTimeRangeId !== id && range.overlappedRangeId !== id,
+    );
+
+    if (overlappedFromRange.length > 0) {
+      showOverlapMessage();
+      overlappedFromRange.forEach((range) => {
+        if (
+          !newOverlappedRanges.some(
+            (currentRange) =>
+              currentRange.selectedTimeRangeId === id && currentRange.overlappedRangeId === range.id,
+          )
+        ) {
+          newOverlappedRanges.push({ selectedTimeRangeId: id, overlappedRangeId: range.id });
+        }
+      });
+    }
+
+    if (overlappedToRange.length > 0) {
+      showOverlapMessage();
+      overlappedToRange.forEach((range) => {
+        if (
+          !newOverlappedRanges.some(
+            (currentRange) =>
+              currentRange.selectedTimeRangeId === id && currentRange.overlappedRangeId === range.id,
+          )
+        ) {
+          newOverlappedRanges.push({ selectedTimeRangeId: id, overlappedRangeId: range.id });
+        }
+      });
+    }
+
+    store.dispatch(areaAndTimeSectionSlice.actions.setRangesOverlapped(newOverlappedRanges));
+
+    removeOverlapFlag(newOverlappedRanges);
+  };
+
+  const showOverlapMessage = () => {
+    store.dispatch(
+      notificationSlice.actions.displayPanelError({
+        message: t`Please select non-overlapping time ranges.`,
+        canBeClosed: false,
+      }),
+    );
+  };
+
+  const removeOverlapFlag = (newOverlappedRanges) => {
```

Comment:
Function name might be misleading, since it removes the error and not the flag (better name then removeOverlapError()). 

Also it will only remove the error if there are no overlaps anymore, so calling removeOverlapError() might not actually always remove it. I would either rename the function to something like removeErrorIfNoOverlap() or "reverse" the logic and extract the if outside of the function and then call it

if (newOverlappedRanges.length === 0) {
   removeOverlapError();
}

const removeOverlapError = () => {
   store.dispatch(...)
}